himbaechel/arch.cc

```diff
@@ -504,4 +504,61 @@ IdString Arch::get_tile_type(int tile) const
     return IdString(tile_data.type_name);
 }
 
+std::vector<GraphicElement> Arch::getDecalGraphics(DecalId decal) const
+{
+    std::vector<GraphicElement> ret;
+    if (decal.type == DecalId::TYPE_BEL) {
+        BelId bel(decal.tile, decal.index);
+        GraphicElement::style_t style = decal.active ? GraphicElement::STYLE_ACTIVE : GraphicElement::STYLE_INACTIVE;
+        uarch->drawBel(ret, style, getBelType(bel), getBelLocation(bel));
+    } else if (decal.type == DecalId::TYPE_WIRE) {
```

Comment:
if a wire spans multiple tiles, as it can in the himbaechel deduplication scheme, this would need to deal with all the constituent tile wires not just the index one

Reply:
@gatecat other points are addressed, except this one. Is there already some other part of code where there is also looping the list of constituent tile wires ? if so would like to be consistent.